src/terraform-resource/terraform/client.go

```diff
@@ -110,6 +115,51 @@ func (c Client) Destroy() error {
 	return nil
 }
 
+
+func (c Client) Plan() error {
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "terraform-resource-client")
+	if err != nil {
+		return fmt.Errorf("Failed to create temporary working dir at '%s'", os.TempDir())
+	}
+	defer os.RemoveAll(tmpDir)
+
+	initCmd := terraformCmd([]string{
+		"init",
+		c.Model.Source,
+		tmpDir,
+	})
+	if initOutput, initErr := initCmd.CombinedOutput(); initErr != nil {
+		return fmt.Errorf("terraform init command failed.\nError: %s\nOutput: %s", initErr, initOutput)
+	}
+
+	getCmd := terraformCmd([]string{
+		"get",
+		"-update",
+		tmpDir,
+	})
+	if getOutput, getErr := getCmd.CombinedOutput(); getErr != nil {
+		return fmt.Errorf("terraform get command failed.\nError: %s\nOutput: %s", getErr, getOutput)
+	}
+
+	planArgs := []string{
+		"plan",
+		"-input=false", // do not prompt for inputs
+		"-out=plan",
```

Comment:
As a second pass, we should make sure to clean up plan files we no longer need from S3. For example, when the `apply` succeeds we could remove any plan files for that environment.